### DISCUSSION TITLE
Is there some convenient way to use a Helm chart from a specific commit or some dev version?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [apache/airflow#45293] [Q&A] [answered]
Related to https://github.com/apache/airflow/discussions/44813.
It would be helpful to have some Helm chart changes being used when trying to configure a self-hosted Airflow. There are some instructions about setting up a chart repo, but that seems cumbersome, so I wonder if there's something already in place for this.
Alternatively, how hard is it to create 1.15.1 with some changes cherry picked and release it soon? I'm happy to contribute PRs to do that if that's an option.

**Comment by potiuk** [comment, marked as answer]
You can release the chart from main version yourself - it's just a matter of following the steps there https://github.com/apache/airflow/blob/main/dev/README_RELEASE_HELM_CHART.md -> and publishing it somewhere you can install it from (local folder if you are self-hosting it is enough).

When it comes to releases - best place to go and ask for it interactively is #release-management channel in Airfow Slack - I guess it's a matter of having a number of other things and Airflow 3 preparation, and holiday period does not help - and also some of the thinking around breaking/non-breaking  release and testing quite some testing. If think the best way to help there is not to have PRs but mostly to do some manual testing of what is already there and confirming that all the features added since the last release are working, the release process itself is easy, but having at lest rudimentary check on all the features added so far and reporting "this has been testeid - this is the result" is one of the best ways to make the release faster.

Would you agree on that @jedcunningham ? And @andrii-korotkov-verkada - would you be fine with spending some effort on testing it?

**Reply by andrii-korotkov-verkada** [reply]
Yeah, I'll spend some effort on testing things, would probably get a summary of things to check by looking at diffs in helm charts since 1.15.0 and see what features need to be tested.

**Reply by jedcunningham** [reply]
I plan to start working on the next release soon here, been busy with Airflow 3. And yes, there isn't much to do to help, but testing main is always helpful. It's rare I don't notice something that needs to be adjusted before we can release - if that happens ahead of time it's a bit easier on me.

@potiuk definitely kicking the can on breaking changes for now.

**Reply by andrii-korotkov-verkada** [reply]
For adhoc chart hosting in my Github repo, do I understand correctly that I can just follow https://helm.sh/docs/topics/chart_repository/#github-pages-example and then commit and push `index.html` and generated `*.tgz.*` artifacts into my branch?

**Reply by potiuk** [reply]
I don't think you need index. We simply publish it following release process, I guess just publishing the right .tar.gz file via any webserver should be good https://downloads.apache.org/airflow/helm-chart/1.15.0/

But I am not 100% sure.

**Reply by jedcunningham** [reply]
You can also just point helm at your local airflow clone too. I almost always do it that way. No need to package it.

**Reply by andrii-korotkov-verkada** [reply]
I'll probably package it, since I wanna it to be reproducible if someone else wants to check out the code and do some updates.

**Reply by andrii-korotkov-verkada** [reply]
Seems like the minimal setup for publishing as GH page is having `*.tgz` and `index.html` files in the root of the repo in `gh-pages` branch. Then just specifying the repo url in the chart like `https://andrii-korotkov-verkada.github.io/airflow`
